src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,169 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private static final int INVALID_INDEX = -1;
+    private Node<T> head;
+    private Node<T> tail;
+    private int size = 0;
+
+    private static class Node<E> {
+        private E item;
+        private Node<E> previous;
+        private Node<E> next;
+
+        public Node(Node<E> previous, E item, Node<E> next) {
+            this.item = item;
+            this.previous = previous;
+            this.next = next;
+        }
+    }
```

Comment:
Private methods and classes should be after public ones in your class. Move the `Node` class to the end of `MyLinkedList` class.